core/src/stdlib/pyscript/events.py

```diff
@@ -1,134 +1,235 @@
+
+    # Attach the same handler to multiple events but not as a decorator.
+    when([my_event, another_event])(another_handler)
+
+    # Trigger an Event instance from a DOM event via @when.
+    @when("click", "#my-button")
+    def handle_click(event):
+        another_event.trigger("Button clicked!")
+
+    # Stacked decorators also work.
+    @when("mouseover", "#my-div")
+    @when(my_event)
+    def handle_both(event):
+        display("Either mouseover or custom event triggered!")
+    ```
     """
-    # If "when" is called as a function, try to grab the handler from the
-    # arguments. If there's no handler, this must be a decorator based call.
-    handler = None
-    if args and (callable(args[0]) or is_awaitable(args[0])):
-        handler = args[0]
-    elif callable(kwargs.get("handler")) or is_awaitable(kwargs.get("handler")):
-        handler = kwargs.pop("handler")
-    # If the target is a string, it is the "older" use of `when` where it
-    # represents the name of a DOM event.
-    if isinstance(target, str):
-        # Extract the selector from the arguments or keyword arguments.
-        selector = args[0] if args else kwargs.pop("selector")
+    if isinstance(event_type, str):
+        # This is a DOM event to handle, so check and use the selector.
         if not selector:
-            msg = "No selector provided."
-            raise ValueError(msg)
-        # Grab the DOM elements to which the target event will be attached.
-        from pyscript.web import Element, ElementCollection
-
-        if isinstance(selector, str):
-            elements = document.querySelectorAll(selector)
-        elif isinstance(selector, Element):
-            elements = [selector._dom_element]
-        elif isinstance(selector, ElementCollection):
-            elements = [el._dom_element for el in selector]
-        else:
-            elements = selector if isinstance(selector, list) else [selector]
+            raise ValueError("Selector required for DOM event handling.")
+        elements = _get_elements(selector)
+        if not elements:
+            raise ValueError(f"No elements found for selector: {selector}")
 
     def decorator(func):
-        sig = inspect.signature(func)
-        if sig.parameters:
-            if is_awaitable(func):
+        wrapper = _create_wrapper(func)
+        if isinstance(event_type, Event):
+            # Custom Event - add listener.
+            event_type.add_listener(wrapper)
+        elif isinstance(event_type, list) and all(
+            isinstance(t, Event) for t in event_type
+        ):
+            # List of custom Events - add listener to each.
+            for event in event_type:
+                event.add_listener(wrapper)
+        else:
+            # DOM event - attach to all matched elements.
+            for element in elements:
+                element.addEventListener(
+                    event_type, create_proxy(wrapper), to_js(options)
```

Comment:
argh! sorry about that ... if options has no key you should *not* pass it or pass `False` instead, otherwise is handled as *truthy* which is a signature overload of `addEventListener` , see [useCapture](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#syntax)

Reply:
OK. So I've done:

```python
element.addEventListener(
    event_type,
    create_proxy(wrapper),
    to_js(options) if options else False,
)
```

So `False` is passed if there are no `**options`.